android/BOINC/app/src/main/java/edu/berkeley/boinc/attach/ProjectAttachService.java

```diff
@@ -372,6 +371,9 @@ public int attachAcctMgr(String url, String name, String pwd) {
         AcctMgrInfo info = null;
         try {
             info = monitor.getAcctMgrInfo();
+            if(info.acct_mgr_url.equals("") && info.acct_mgr_name.equals("") && !info.have_credentials){
+                return  BOINCErrors.ERR_BAD_PASSWD;
+            }
```

Comment:
While in common check is fine, in real case this error message could confuse user because it's too general.
E.g. user has no account on Account Manager but will receive "bad password" error. Usually the server send back an information what exactly went wrong, and this could be done via `acct_mgr_rpc_poll` RPC. Please use detailed description here: https://boinc.berkeley.edu/trac/wiki/GuiRpcProtocol#acct_mgr_rpc